File: DSA/Manav-Rajput/Day42_question2.java
class Solution {
    public int maxDistToClosest(int[] seats) {
        int n = seats.length;
        int maxDist = 0;
        int prevOccupied = -1;

        for (int i = 0; i < n; i++) {
            if (seats[i] == 1) {
                if (prevOccupied == -1) {
                    maxDist = i;
                } else {
                    maxDist = Math.max(maxDist, (i - prevOccupied) / 2);
                }
                prevOccupied = i;
            }
        }
        maxDist = Math.max(maxDist, n - 1 - prevOccupied);

        return maxDist;
    }
}

File: DSA/Arshdeep-Kaur/Day 42/Day42_Question2.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int ans{}, count{};
        bool isFound{false};
        for(int i{}; i < (int)seats.size(); ++i){
            if(seats[i] == 0)
                ++count;
            else if(seats[i] == 1){
                if(isFound)
                    ans = max(ans, (count + 1) / 2);
                else
                    ans = max(ans, count);
                isFound = true;
                count = 0;
            }
        }
        if(count)
            ans = max(ans, count);
        return ans;
    }
};
